### ISSUE TITLE
Output freezes

### ISSUE THREAD

**Issue by ofwinterpassed** [djpohly/dwl#115]
Hi when I'm using dwl sometimes the whole wayland output stops updating. Switching to a virtual terminal and back again unfreezes it.

**Comment by novakne** [comment]
Yeah that's a bug that all wlroots compositor have, the only difference is compositor with damage tracking doesn't have freeze when this bugs occurs

**Comment by djpohly** [comment]
@novakne Interesting. What part of the damage tracking code handles the pageflip error?

**Comment by novakne** [comment]
Tbh I don't know exactly why damage tracking hide this bug, It seems that damage tracking scheduled a new frame even whith output commit failure. That's why there is a workaround like, take a screenshot every second or so. Not great on the cpu though

I found this bug in dwl, river, sway and even tinywl so it looks like its a bug from wlroots drm backend but I don't have the knowledge to debug this
Also looks like that people with an integrated intel graphics chip, and maybe laptop are the most affected

**Comment by thavelick** [comment]
Looks like this bug is being tracked by the wlroots team here: https://gitlab.freedesktop.org/wlroots/wlroots/-/issues/2937. Though there wasn't been any action over there in quite a while

**Comment by Sevz17** [comment]
Hey @ofwinterpassed, with scenegraph we have damage tracking, is it still happening?

**Comment by ofwinterpassed** [comment]
Sorry for the slow answer. I'm haven't had time to look into this for some time. I've used damage tracking in my build for some time and that took care of the problem up until I integrated latest master a couple of months ago which made me try without the changes for damage tracking. On that build I have definitely had very few freezes and the once I've had has been different (I used to be able to switch to a terminal and back to dwl and it unfroze. Now it's completely frozen and the only way to get out of it is the powerbutton that triggers a shutdown)

**Comment by gamodri** [comment]
I'm running dwl wlroots-next branch with wlroots git, I pull/rebuild mostly daily, and I'm experiencing the nastier version of freezes described by @ofwinterpassed: unable to recover the session switching to vt and back, simply freezed.

I'm running an home-brewed aarch64 CRUX Linux on a pinebook pro, with a bit outdated toolchain (glibc 2.32, gcc 10.4.0, binutils 2.35.2).

I've experienced the same issue with identical behavior, even before switching to the CRUX rootfs, on the stock Manjaro with dwl master branch and wlroots 0.15.1

I started using wayland/dwl a pair of weeks ago, no other wayland/wlroots experience before, so I don't precisely know what to do to provide more informations about this issure, I'd be glad to help (dwl rocks).

**Comment by leg7** [comment]
Is there something I can do to fix/patch this? Currently dwl freezes every 30m/1h and I have to power off my system to stop it. I've tried running the `wlroots-next` branch with wlroots-git but it doesn't change anything.

**Comment by sevz17** [comment]
@leg7, it is likely a different issue, can you check https://github.com/djpohly/dwl#status-information and try what it says.
